### PR TITLE
feat: direct grants simple event handlers

### PR DESCRIPTION
**Pull request by 0xnigir1**
# 🤖 Linear

Closes GIT-161 GIT-163 

## Description
- add `DirectGrantsSimpleStrategy` handler and events handlers:
  - `Registered`
  - `TimestampsUpdated`
  - `Distributed`

## Checklist before requesting a review

-   [x] I have conducted a self-review of my code.
-   [x] I have conducted a QA.
-   [x] If it is a core feature, I have included comprehensive tests.

### CODE REVIEW

File: packages/processors/src/processors/strategy/directGrantsSimple/handlers/registered.handler.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ *
+ * This handler performs the following core actions when a project registers for a round:
+ * - Validates that both the project and round exist
+ * - Decodes the application data from the event
+ * - Retrieves the application metadata
+ * - Creates a new application record with PENDING status
+ * - Links the application to both the project and round
+ */
+
+export class DGSimpleRegisteredHandler
```
Comment (0xkenj1) [inline packages/processors/src/processors/strategy/directGrantsSimple/handlers/registered.handler.ts:25]:
```suggestion
export class DirectGrantsSimpleRegisteredHandler
```

may we use complete names for strategies ? or at least use the same for strategy and event handlers :) 

Look here packages/processors/src/processors/strategy/directGrantsSimple/directGrantsSimple.handler.ts

**Review by 0xkenj1 — DISMISSED** [review]

File: packages/processors/test/strategy/directGrantsSimple/handlers/registered.handler.spec.ts [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    Round,
+    RoundNotFound,
+} from "@grants-stack-indexer/repository";
+import { ChainId, ProcessorEvent } from "@grants-stack-indexer/shared";
+
+import { DGSimpleRegisteredHandler } from "../../../../src/processors/strategy/directGrantsSimple/handlers/registered.handler.js";
+import { createMockEvent } from "../../../mocks/index.js";
+
+// function createMockEvent(
+//     overrides: DeepPartial<ProcessorEvent<"Strategy", "RegisteredWithSender">> = {},
```
Comment (jahabeebs) [inline packages/processors/test/strategy/directGrantsSimple/handlers/registered.handler.spec.ts]: commented code

**Review by jahabeebs — DISMISSED** [review]
small code cleanup comment but looks good 👍🏻

File: packages/processors/test/strategy/directGrantsSimple/handlers/registered.handler.spec.ts (hunk 2 above)
Comment (0xnigir1) [reply packages/processors/test/strategy/directGrantsSimple/handlers/registered.handler.spec.ts]: [9d6a824](https://github.com/defi-wonderland/grants-stack-indexer-v2/pull/35/commits/9d6a824dd6f7556b247ec858885b8ac1ed1a6194)

File: packages/processors/src/processors/strategy/directGrantsSimple/handlers/registered.handler.ts (hunk 1 above)
Comment (0xnigir1) [reply packages/processors/src/processors/strategy/directGrantsSimple/handlers/registered.handler.ts:25]: [9d6a824](https://github.com/defi-wonderland/grants-stack-indexer-v2/pull/35/commits/9d6a824dd6f7556b247ec858885b8ac1ed1a6194)

i renamed to DGSimple to follow the convention on the other strategies

**Review by 0xkenj1 — APPROVED** [review]

**Review by jahabeebs — APPROVED** [review]

**Pull request merged**